kubernetes/samples/scripts/domain-lifecycle/stopServer.sh

```diff
@@ -0,0 +1,124 @@
+  Usage:
+ 
+    $(basename $0) -s myserver [-n mynamespace] [-d mydomainuid]
+  
+    -s <server_name>           : Server name parameter is required.
+
+    -d <domain_uid>            : Default is 'sample-domain1'.
+
+    -n <namespace>             : Default is 'sample-domain1-ns'.
+    
+    -k <keep_replica_constant> : Keep replica count constant. Default behavior is to decrement replica count.
+
+    -m <kubernetes_cli>        : Kubernetes command line interface. Default is 'kubectl'.
+
+    -h                         : This help.
+   
+EOF
+exit $1
+}
+
+kubernetesCli=${KUBERNETES_CLI:-kubectl}
+serverName=""
+clusterName=""
+domainUid="sample-domain1"
+domainNamespace="sample-domain1-ns"
+keepReplicaConstant=false
+
+while getopts "ks:m:n:d:h:" opt; do
+  case $opt in
+    s) serverName="${OPTARG}"
+    ;;
+    n) domainNamespace="${OPTARG}"
+    ;;
+    m) kubernetesCli="${OPTARG}"
+    ;;
+    d) domainUid="${OPTARG}"
+    ;;
+    k) keepReplicaConstant=true;
+    ;;
+    h) usage 0
+    ;;
+    *) usage 1
+    ;;
+  esac
+done
+
+#
+# Function to perform validations, read files and initialize workspace
+#
+function initialize {
+
+  # Validate the required files exist
+  validateErrors=false
+
+  validateKubernetesCliAvailable
+  validateJqAvailable
+
+  if [ -z "${serverName}" ]; then
+    validationError "Please specify name of server to start using '-s' parameter e.g. '-s managed-server1'."
+  fi
+
+  failIfValidationErrors
+}
+
+initialize
+
+# Get the cluster name for current server 
+clusterName=$(${kubernetesCli} get pod ${domainUid}-${serverName} -n ${domainNamespace} -o=jsonpath="{.metadata.labels['weblogic\.clusterName']}")
+
+# Get the domain in json format
+domainJson=$(${kubernetesCli} get domain ${domainUid} -n ${domainNamespace} -o json)
+
+# Create server start policy patch with IF_NEEDED value
+serverStartPolicy=NEVER
+createServerStartPolicyPatch "${domainJson}" "${serverName}" "${serverStartPolicy}" serverStartPolicyPatch policy
+
+if [[ -n "${clusterName}" && "${keepReplicaConstant}" != 'true' ]]; then
+  # if server is part of a cluster and replica count needs to be updated, update replica count and patch server start policy
+  operation="DECREMENT"
+  createReplicaPatch "${domainJson}" "${clusterName}" "${operation}" replicaPatch replicaCount
+  patchJson="{\"spec\": {\"clusters\": "${replicaPatch}",\"managedServers\": "${serverStartPolicyPatch}"}}"
+  echo "[INFO] Patching start policy of server '${serverName}' from '${policy}' to 'NEVER' and decrementing replica count for cluster '${clusterName}'."
+else
+  # if server is an independent managed server or replica count needs to stay constant, only patch server start policy
+  patchJson="{\"spec\": {\"managedServers\": "${serverStartPolicyPatch}"}}"
+  echo "[INFO] Patching start policy of '${serverName}' from '${policy}' to 'NEVER'."
+fi
+${kubernetesCli} patch domain ${domainUid} -n ${domainNamespace} --type='merge' --patch "${patchJson}"
+
+if [ $? != 0 ]; then
```

Comment:
'set -e' is called above, so I assume the $? question check isn't needed?

Reply:
Yes, I have removed the $? check. Thanks.